opendatahub-io/odh-dashboard
# Update 'Resource name' fields to meet UX guidelines: Model registries and Inference services

Towards [RHOAIENG-14746](https://issues.redhat.com/browse/RHOAIENG-14746)

## Description
Updates create/edit modals to use `K8sNameDescriptionField` component in order to meet current UX guidelines.
Modals updated:

- Model Registries
- Project Serving (these share the same name)
  - Serving Runtimes (project “model servers”)
  - Inference Services (project models)

## Screen shots

### Deploy model
![image](https://github.com/user-attachments/assets/218be3ea-97fe-4f36-b122-6d98892f46c0)

### Edit model
![image](https://github.com/user-attachments/assets/e6b99925-b4f1-432a-b522-b880cd13e1e4)

### Model registry - Deploy model
![image](https://github.com/user-attachments/assets/9eddea37-7778-4a40-8c50-6f8277da6cc1)

## How Has This Been Tested?
- Navigate to the Model Serving section
  - Select a project
  - Select `Deploy model`
  - Enter the model deployment name
  - View the resource name via the `Edit resource name` link.
  - Change the resource name to something invalid (ie. contains a space)
  - Verify the submit button is disabled
  - Change the resource name to something valid
  - Verify the submit button is enabled
  - Save the deployment.
  - Verify the deployment is created with the correct Display name and Resource Name.

- Navigate to the Model Serving section
  - For an existing Model, select the kebab and choose `Edit`
  - Update the Name field
  - Notice the resource name is not editable
  - Save the model server.
  - Verify the model is updated with the correct Display name.

- Navigate to the Model Registry section
  - Select an existing model
  - For a version, select the kebab and then `Deploy`
  - Select a project
  - Verify the model deployment name is set
  - Enter data for the rest of the form such that `Deploy` is enabled
  - View the resource name via the `Edit resource name` link.
  - Change the resource name to something invalid (ie. contains a space)
  - Verify the `Deploy` button is disabled
  - Change the resource name to something valid
  - Verify the `Deploy` button is enabled
  - Save the deployment.
  - Navigate to the Model Serving section
  -  Find the new model created, select the kebab and choose `Edit`
  - Verify the deployment was created with the correct Display name and Resource Name.

- Navigate to a project that is not configured for kserve or model mesh
  - Select `Select single-model` from the Single-model serving platform card
  - Select `Deploy model`
  - Enter the model deployment name
  - Enter data for the rest of the form such that `Deploy` is enabled
  - View the resource name via the `Edit resource name` link.
  - Change the resource name to something invalid (ie. contains a space)
  - Verify the `Deploy` button is disabled
  - Change the resource name to something valid
  - Verify the `Deploy` button is enabled
  - Save the deployment.
  - Navigate to the Model Serving section
  -  Find the new model created, select the kebab and choose `Edit`
  - Verify the deployment was created with the correct Display name and Resource Name.


- Navigate to a project this model servers and select the `Models` tab
  - Select `Add model server`
  - Enter the name for the new model server.
  - View the resource name via the `Edit resource name` link.
  - Change the resource name to something invalid (ie. contains a space)
  - Verify the submit button is disabled
  - Change the resource name to something valid
  - Verify the submit button is enabled
  - Save the profile.
  - Verify the profile is created with the correct Display name and Resource Name.
- Navigate to a project this model servers and select the `Models` tab
  - Select the kebab for an existing model server and choose `Edit`
  - Update the Name field
  - Notice the resource name is not editable
  - Save the model server.
  - Verify the model server is added with the correct Display name and resource name

## Test Impact
Updated unit tests for new fields

## Request review criteria:

/cc @simrandhaliw
